libs/ui/search/src/lib/results-table/results-table.component.ts

```diff
@@ -47,12 +40,16 @@ export class ResultsTableComponent {
   @Input() records: CatalogRecord[] = []
   @Input() selectedRecordsIdentifiers: string[] = []
   @Input() sortOrder: SortByField = null
-  @Input() recordHasDraft: (record: CatalogRecord) => boolean = () => false
+  @Input() hasDraft: (record: CatalogRecord) => boolean = () => false
+  @Input() isDraft: (record: CatalogRecord) => boolean = () => false
```

Comment:
After thinking about it, I feel like these two distinct functions might be confusing. When we're showing a list of records, we're showing _records_ and nothing else; drafts are not another type of records, they are only a bunch of pending changes that were not saved yet.

As such, a record would only have two different "flags":
* "hasDraft" means the record has pending changes that were not saved yet, and
* "notSavedYet" means the record was never saved (i.e. equivalent to "isDraftWithoutRecord")

I think it's important here to not give drafts an sort of existence on their own as a separate type of entity, otherwise it's going to be confusing in the long run. What do you think?

Reply:
As for the fact that we currently don't allow deleting a draft from the draft menu if it has been saved already, I would go with a simple `allowDelete: (record) => bool` optional input that would always return true by default. I feel like this might be temporary anyway since we'll need to find a way to handle this case when rolling back will be implemented.

Reply:
I'll try moving the `canDuplicate` and `canDelete` to the inputs of the `results-table`.

Reply:
yes that's what I meant, sorry that wasn't super clear